packages/eslint-plugin/src/rules/no-restricted-imports.ts

```diff
@@ -27,18 +27,18 @@ const allowTypeImportsOptionSchema = {
 const schemaForMergeArrayOfStringsOrObjects = {
   items: {
     anyOf: [
-      {},
```

Comment:
These seem incorrect and unnecessary. I think the intention was for this to merge with the base `anyOf` items, but deepMerge doesn't handle array items: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/utils/src/eslint-utils/deepMerge.ts

Although thinking about this now, I'm not certain the updated rule is 100% correct, would appreciate careful eyes on it. Might need to refactor how this extends from the base.

Reply:
Have redone this. It's fairly horrible how it indexes into the baseRule - happy to take feedback on better approaches.

Reply:
Yeah seems like you're certainly following the standard practice: https://github.com/eslint/eslint/blob/1fea2797801a82a2718814c83dad641dab092bcc/lib/rules/no-restricted-imports.js#L19.

You might want to cast `const baseSchema = baseRule.meta.schema as SomeTypeDescribingWhatWeKnow`, since this is making an assumption on the base rule's schema type.

Reply:
Good idea, have done this.